docs: health check status description (#10547)
apache/apisix: docs/en/latest/tutorials/health-check.md

curl http://127.0.0.1:9090/v1/healthcheck/upstreams/healthycheck -s | jq .

```

> ## Health Check Information

Reviewer: ```suggestion
## Health Check Status
```

> APISIX provides comprehensive health check information, with particular emphasis on the `status` and `counter` parameters for effective health monitoring. In the APISIX context, nodes exhibit four states: `healthy`, `unhealthy`, `mostly_unhealthy`, and `mostly_healthy`. The transition of a node's state depends on the success or failure of the current health check, along with the recording of four key metrics in the `counter`: `tcp_failure`, `http_failure`, `success`, and `timeout_failure`.

Reviewer: I suggest we _explicitly_ explain `mostly_healthy` and `mostly_unhealthy` as well. For example, `mostly_healthy` status means the current node status is healthy, but APISIX starts to receive unhealthy indications during health checks.

I understand this concept is somewhat conveyed in the state transition diagram. However, I find it a bit easier to understand if you explain it up front as well. What do you guys think? @luoluoyuyu @monkeyDluffy6017

To retrieve health check information, you can use the following curl command:

```shell
curl -X GET http://your_apisix_server/your_health_check_endpoint
```

Response Example:

```json
[
  {
    "nodes": {},
    "name": "/apisix/routes/1",
    "type": "http"
  },
  {
    "nodes": [
      {
        "port": 1970,
        "hostname": "127.0.0.1",
        "status": "healthy",
        "ip": "127.0.0.1",
        "counter": {
          "tcp_failure": 0,
          "http_failure": 0,
          "success": 0,
          "timeout_failure": 0
        }
      },
      {
        "port": 1980,
        "hostname": "127.0.0.1",
        "status": "healthy",
        "ip": "127.0.0.1",
        "counter": {
          "tcp_failure": 0,
          "http_failure": 0,
          "success": 0,
          "timeout_failure": 0
        }
      }
    ],
    "name": "/apisix/routes/example-hc-route",
    "type": "http"
  }
]
```

> ### State Diagram

Reviewer: ```suggestion
### State Transition Diagram
```

![image](../../../assets/images/health_check_node_state_diagram.png)

> All nodes start with an initial status of `healthy`. It's important to note that when a node is in the `healthy` state and the health check succeeds, or if the node is in the `unhealthy` state and the health check fails, the data in the counter will not be updated. The counter is only refreshed under different circumstances.

Reviewer: ```suggestion
Note that all nodes start with the `healthy` status without any initial probes, and the counter only resets and updates with a state change. Hence, when nodes are `healthy` and all subsequent checks are successful, the `success` counter is not updated and remains zero.
```

This suggested change is more on the technical fact than writing. You might want to re-organize a bit as there is a `Counter Information` section below.

I removed the explanation of `if the node is in the unhealthy state and the health check fails, the data in the counter will not be updated.` because if the node is in unhealthy state, there has to be a state transition prior. As a result, the counter would actually update as one would expected.
